controllers/searchoperator_controller.go

```diff
@@ -631,32 +660,53 @@ func updateCustomizationCR(kclient client.Client, cr *searchv1alpha1.SearchCusto
 	return nil
 }
 
-func updateRedisStatefulSet(client client.Client, deployment *appv1.StatefulSet) {
+func statefulSetNeedsUpdate(client client.Client, deployment *appv1.StatefulSet) bool {
 	found := &appv1.StatefulSet{}
 	err := client.Get(context.TODO(), types.NamespacedName{Name: statefulSetName, Namespace: namespace}, found)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			err = client.Create(context.TODO(), deployment)
-			if err != nil {
-				log.Error(err, "Failed to create  deployment")
+		return true
+	} else {
+		if !reflect.DeepEqual(found.Spec, deployment.Spec) ||
+			!reflect.DeepEqual(found.GetObjectMeta(), deployment.GetObjectMeta()) {
+			log.Info("Volume source and/or metadata needs to be updated for redisgraph Statefulset")
+			return true
+		} else {
+			log.Info("No changes required for Statefulset")
+			return false
+		}
+	}
+}
+func updateRedisStatefulSet(client client.Client, deployment *appv1.StatefulSet) {
+	createUpdateSTS := statefulSetNeedsUpdate(client, deployment)
+	if !createUpdateSTS {
```

Comment:
can we check statefulSetNeedsUpdate - before we call updateRedisStatefulSet ? Now we call update , then come in the function and check if we needs update .

Reply:
Done